src/encoded/static/components/regulome_search.js

```diff
@@ -1252,14 +1272,50 @@ class RegulomeSearch extends React.Component {
     render() {
         const context = this.props.context;
         const urlBase = this.context.location_href.split('/regulome-search')[0];
-        const coordinates = Object.keys(context.variants)[0];
+        const coordinates = context.query_coordinates[0];
         const allData = context['@graph'] || [];
         const QTLData = allData.filter(d => (d.method && d.method.indexOf('QTL') !== -1));
         const chipData = allData.filter(d => d.method === 'ChIP-seq');
         const dnaseData = allData.filter(d => (d.method === 'FAIRE-seq' || d.method === 'DNase-seq'));
         const chromatinData = allData.filter(d => (d.method === 'chromatin state'));
         const thumbnail = this.context.location_href.split('/thumbnail=')[1] || null;
 
+        const toggleFreqsShow = () => this.setState(
+            state => ({ showMoreFreqs: !state.showMoreFreqs })
+        );
+        const hitSnps = {};
+        const sortedPopulations = {};
+        if (coordinates) {
+            const [chrom, startEnd] = coordinates.split(':');
+            const [start, end] = startEnd.split('-');
+            context.nearby_snps.forEach((snp) => {
+                if (snp.chrom === chrom && snp.coordinates.gte === +start && snp.coordinates.lt === +end) {
+                    hitSnps[snp.rsid] = {};
+                    if (snp.ref_allele_freq) {
+                        Object.keys(snp.ref_allele_freq).forEach((allele) => {
+                            Object.keys(snp.ref_allele_freq[allele]).forEach((population) => {
+                                hitSnps[snp.rsid][population] = `${allele}=${snp.ref_allele_freq[allele][population]}`;
+                            });
+                        });
+                    }
+                    if (snp.alt_allele_freq) {
+                        Object.keys(snp.alt_allele_freq).forEach((allele) => {
+                            Object.keys(snp.alt_allele_freq[allele]).forEach((population) => {
+                                if (!hitSnps[snp.rsid][population]) {
+                                    hitSnps[snp.rsid][population] = `${allele}=${snp.alt_allele_freq[allele][population]}`;
+                                } else {
+                                    hitSnps[snp.rsid][population] += `, ${allele}=${snp.alt_allele_freq[allele][population]}`;
```

Comment:
why not use join? Here is an example:
['a', 'b'].join(' , ')
As in python, in Javascript, strings are immutatable. So, you are constantly creating a new string and its a slow process.

Reply:
Thanks for explaining that. Make sense. I can build an array and join after the iteration.